hedgehog/src/Hedgehog/Internal/Report.hs

```diff
@@ -715,14 +718,58 @@ ppResult name (Report tests discards result) =
         ppDiscardCount discards <>
         ", passed" <+>
         ppTestCount tests <>
-        "."
+        "." <+>
+        ppClassification classes <+>
+        ppCoverage classes
 
     OK ->
       pure . icon SuccessIcon '✓' . WL.annotate SuccessHeader $
         ppName name <+>
         "passed" <+>
         ppTestCount tests <>
-        "."
+        "." <+>
+        ppClassification classes <+>
+        ppCoverage classes
+
+ppClassification :: Classification -> Doc Markup
+ppClassification (Classification classifiers total) =
+  if Map.null classifiers then
+    mempty
+  else
+    (<>) WL.linebreak $ WL.indent 4 . WL.align . WL.vsep $
+      (\(ClassifierName k, v) -> WL.text $ show (occurrenceRate v total) <> "% " <> k) <$> Map.toList classifiers
+
+occurrenceRate :: Classifier -> Integer -> Double
+occurrenceRate (Classifier _ occurrences) total =
+  let
+    percentage :: Double
+    percentage =
+      fromIntegral occurrences / fromIntegral (total - 2) * 100
+    thousandths :: Integer
+    thousandths =
+      round $ percentage * 10
+  in
+    fromIntegral thousandths / 10
```

Comment:
There's a lot of number fudging here. What's the purpose?

Reply:
Rounding to 1 decimal place

Reply:
There's also a minus 2?

Reply:
I think if we can get rid of those magic numbers—or at least make it more clear for the code reader (remember, code is read more often than it is written)—perhaps we can make some progress on this? /cc @jystic 

On next PRs we can then
* deal with the statistical stuff as commented by @HuwCampbell [here](https://github.com/hedgehogqa/haskell-hedgehog/pull/253/files#r267271586)
* refactor `(Classification, [Log])` into a `Journey` type

Reply:
So when I run this, it always seems to give an off by two error in the amount of total tests. I can't figure out where it comes from. Hedgehog doesn't seem to be reporting any discarded tests and I don't think the `Monoid` instance for `Classification` is wrong either...

So either:

1. I don't completely understand the mtl layering and if it has any adverse effects on the total number of tests...or:
2. There is a bug in hedgehog somewhere else

🤷‍♂️

This is the output that hedgehog generates if I add the total to the prints:

```
Test suite test: RUNNING...
━━━ Test.Hedgehog.Text ━━━
  ✓ prop_show_append_size passed 100 tests. Total: 19
  ✓ prop_tripping_append_size passed 100 tests. Total: 10
  ✓ prop_show_append_seed passed 100 tests. Total: 19
  ✓ prop_tripping_append_seed passed 100 tests. Total: 10
  ✓ 4 succeeded.
━━━ Test.Hedgehog.Seed ━━━
  ✓ prop_avoid_pathological_gamma_values passed 1 test. Total: 34
  ✓ 1 succeeded.
━━━ Test.Hedgehog.Classified ━━━
  ✓ prop_check_classifiers passed 1 test. Total: 102
      50.0% big number
      48.0% small number
  ✓ prop_check_coverage passed 1 test. Total: 102
      49.0% big number
      48.0% small number

  ⚠   Only 49.0% big number, but expected 50.0%
      Only 48.0% small number, but expected 50.0%
  ✓ 2 succeeded.
```